File: api_examples/nodejs/connect.js
/**
 * Complex connection with retries and logging.
 * This simulates a production-grade connection to SolMax AI API server.
 */

const http = require('http');
const fs = require('fs');

const MAX_RETRIES = 3;
let attempts = 0;

function logToFile(message) {
    fs.appendFileSync('connection.log', `[${new Date().toISOString()}] ${message}\n`);
}

function pingServer() {
    const options = {
        hostname: '125.233.44.145',
        port: 4123,
        path: '/api/ping',
        method: 'GET',
        timeout: 3000
    };

    logToFile('Attempting to ping the server...');

    const req = http.request(options, res => {
        let data = '';
        res.on('data', chunk => data += chunk);
        res.on('end', () => {
            logToFile('Server responded with: ' + data);
            console.log('Response:', data);
        });
    });

    req.on('timeout', () => {
        req.abort();
        logToFile('Request timed out.');
        retryConnection();
    });

    req.on('error', error => {
        logToFile('Error: ' + error.message);
        retryConnection();
    });

    req.end();
}

function retryConnection() {
    attempts++;
    if (attempts < MAX_RETRIES) {
        logToFile(`Retrying... (${attempts}/${MAX_RETRIES})`);
        setTimeout(pingServer, 2000);
    } else {
        logToFile('Max retries reached. Exiting.');
    }
}

pingServer();

File: api_examples/python/connect.py
# Complex ping script with retries and logging
import requests, time, logging

logging.basicConfig(filename='connect.log', level=logging.INFO)

def ping_server(attempts=0, max_attempts=3):
    url = 'http://125.233.44.145:4123/api/ping'
    try:
        logging.info(f'Trying to ping SolMax server (attempt {attempts+1})...')
        response = requests.get(url, timeout=3)
        logging.info(f'Status: {response.status_code}, Body: {response.text}')
        print('[✓] Response:', response.text)
    except requests.RequestException as e:
        logging.warning(f'Ping failed: {e}')
        if attempts + 1 < max_attempts:
            time.sleep(2)
            ping_server(attempts + 1)
        else:
            logging.error('Max attempts reached. Giving up.')

ping_server()
